### PR TITLE
Topic depscan

### PR DESCRIPTION
**Pull request by jbest84**
@tlwalla1 @CHenrie 

This pull request adds a new grunt task called deps. This task will scan our ES6 in src and build up a dependency graph. The graph is then sorted and a template is used to output to our release.jsb2 file. This means no more messing around with the file order in the release.jsb2 to get a build working right. This means we need to update the release.tmpl file and run `grunt deps` to get the output.

### CODE REVIEW

**Comment by jbest84** [comment]
Here is a working build using it: http://10.33.29.171/mobile-builds/topic-depscan/627/#_login

File: grunt-tasks/grunt-deps.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    grunt.file.setBase(config.cwd);
+    var files = grunt.file.expand(config.files);
+    var graph = new Graph();
+    var nodes = {};
+
+    // Resolves import modules into a relative file path
+    function resolvePath(module, sourceFile) {
+      var config = grunt.config.get('deps');
+      var relative = module[0] === '.';
+      if (relative) {
```
Comment (tlwalla1) [inline grunt-tasks/grunt-deps.js]: The variable relative isn't used beyond the if statement, you could move it into the if statement and remove the variable declaration. Unless this is used for clarity, which is understandable.

File: grunt-tasks/grunt-deps.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        var sourceDir = path.dirname(sourceFile);
+        return path.join(sourceDir, module) + '.js';
+      } else {
+        var parts = module.split('/');
+        var moduleName = parts.shift();
+        var config = config.modules.filter(function(m) {
+          return m.name === moduleName;
+        })[0];
+        var location = config && config.location;
```
Comment (tlwalla1) [inline grunt-tasks/grunt-deps.js]: Same as above. This time the config.location gives understanding to the check already.

File: grunt-tasks/grunt-deps.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    function add(f) {
+      if (nodes[f] || f === null || typeof f === 'undefined') {
+        return nodes[f];
+      }
+
+      nodes[f] = {
+        name: f
+      };
+
+      if (graph.has(nodes[f])) {
```
Comment (tlwalla1) [inline grunt-tasks/grunt-deps.js]: Is this redundant with the if check above? If it's already in nodes[f] it should be in the graph.

**Comment by jbest84** [comment]
Updated build, argos-sdk and argos-saleslogix both using https://github.com/Saleslogix/grunt-argos-deps

**Comment by CHenrie** [comment]
This will be awesome looks good

**Pull request merged**